test/game_test.rb

```diff
@@ -58,4 +59,25 @@ def test_if_coordinates_are_valid_a_player_can_place_ships
     assert_equal true, game.player_submarine.placed?
   end
 
+  def test_game_is_over_when_all_ships_are_sunk
+    game = Game.new
+
+    assert_equal false, game.player_all_sunk?
+
+
+    game.stubs(:player_all_sunk?).returns(true)
+    assert_equal true, game.player_all_sunk?
+
+    game.game_over?
+
+    assert_equal true, game.game_over?
+
+    assert_equal false, game.computer_all_sunk?
+    game.stubs(:computer_all_sunk).returns(true)
```

Comment:
Nice stubbing, maybe we can try and see if we can find a place to use it else where.